### ISSUE TITLE
Exception Raised when Parameterizing Ions with OpenEye Toolkit

### ISSUE THREAD

**Issue by SimonBoothroyd** [openforcefield/openforcefield#305]
## Expected Behavior
Calling `forcefield.create_openmm_system` on a topology which contains ions (Na+, Cl-...) should return the parameterized system when using the OpenEye toolkit.

## Actual Behavior
A TypeError exception is raised when trying to parameterize systems which contain ions when using the OpenEye toolkit backend. The exception is raised on line 1100 of toolkits.py and relates to no conformers having been generated for the ion.

I haven't tested this with RDKit.

## Minimal Reproducible Example
```
from openforcefield.topology import Molecule, Topology
from openforcefield.typing.engines.smirnoff import ForceField

sodium = Molecule.from_smiles('[Na+]')
topology = Topology.from_molecules(sodium)

force_field_path = get_data_filename('forcefield/smirnoff99Frosst.offxml')
force_field = ForceField(force_field_path)

system = force_field.create_openmm_system(topology)
```

## Specifications

  - Version: 0.3.0-branch

**Comment by SimonBoothroyd** [comment]
Ok it looks like this is caused by the `from_openeye` method in the `OpenEyeToolkitWrapper` class discarding conformers which have all of their positions set to be 0.

**Comment by SimonBoothroyd** [comment]
Fixing the issues with the conformers results in a new unrelated issue being caught. The toolkit attempts to charge the ions using `OEAssignCharges`, however this method fails as it doesn't not support `Na+`, and I imagine assigning charges to ions in general.

It seems like a catch for ions should be added in the `compute_partial_charges_am1bcc`, which just assigns the ion a formal charge.

**Comment by j-wags** [comment]
I think we should avoid assigning a monoatomic ion's partial charge to be equal to its formal charge for now. It seems straightforward, but I suspect we may encounter unintended consequences down the line.

Instead, @SimonBoothroyd and I talked, and agreed that, until we get `LibraryCharges` straightened out, we could have a file defining standard ions through the Toolkit API, and hard-coding them the desired partial charges, then those would be passed in as `ForceField.create_openmm_system`'s `charge_from_molecules` kwarg.

**Comment by jchodera** [comment]
> Instead, @SimonBoothroyd and I talked, and agreed that, until we get LibraryCharges straightened out, we could have a file defining standard ions through the Toolkit API, and hard-coding them the desired partial charges, then those would be passed in as ForceField.create_openmm_system's charge_from_molecules kwarg.

I think this is a good short-term solution, but we have to recognize this will make it impossible to make progress in improving parameters for ions or salts in the future. We will ultimately need some way to apply charge models for ions and salts. Perhaps you can create a new issue where we can discuss that?

**Comment by j-wags** [comment]
> We will ultimately need some way to apply charge models for ions and salts.

@jchodera My thought is that `LibraryCharges` will be implemented in the next few months as our long-term solution. Or are you asking about a larger scope than that?

**Comment by dotsdl** [comment]
Hey @trevorgokey, I committed to taking this issue on along with #379 and #381. Is there any progress you'd like to report to help me get started, or is this perhaps complete?

**Comment by trevorgokey** [comment]
No, you can go ahead with all of them.

**Comment by dotsdl** [comment]
For this issue, if we have decided to proceed with a `LibraryCharges` solution, does anyone have preferences on where single-atom ion definitions should go? Should we add a file in [test_forcefields](https://github.com/openforcefield/openforcefield/tree/master/openforcefield/data/test_forcefields), or is it better to define these somewhere in the library itself as a dict or similar?

**Comment by j-wags** [comment]
I agree with your first suggestion -- we should make a file like `ion_charges.offxml` in `test_forcefields/`, containing library charges for the ions, and add tests for applying them. This file should at least cover ions that [Parsley has vdW parameters for](https://github.com/openforcefield/openforcefields/blob/master/openforcefields/offxml/openff-1.1.0.offxml#L345-L353)

**Comment by j-wags** [comment]
In the long run, we may start including these LibraryCharges in the main line of force fields. I've asked on the #smirnoff channel on the project Slack.

**Comment by j-wags** [comment]
> In the long run, we may start including these LibraryCharges in the main line of force fields

Oh, @davidlmobley already requested this, I just lost track: https://github.com/openforcefield/openforcefields/issues/5

**Comment by j-wags** [comment]
We're converging on the solution here being the addition of these ion charges to the main line of force fields. I'm going to close this issue and move discussion about how to get these charges into `openforcefields` to https://github.com/openforcefield/openforcefields/issues/5

**Comment by dotsdl** [comment]
We've addressed this issue in #563 by introducing an `ion_charges.offxml` file in `test_forcefields`. This will need to be included in [openforcefields](https://github.com/openforcefield/openforcefields) in order to be slated for production use. As of that PR, the following now works with `openforcefields` installed:

```python
from openforcefield.topology import Molecule, Topology
from openforcefield.typing.engines.smirnoff import ForceField

sodium = Molecule.from_smiles('[Na+]')
topology = Topology.from_molecules(sodium)

#force_field_path = get_data_filename('forcefield/smirnoff99Frosst.offxml')
force_field = ForceField('openff-1.1.0.offxml',
                         'test_forcefields/ion_charges.offxml')

system = force_field.create_openmm_system(topology)
```
I am closing this issue as the toolkit has everything it needs to proceed. I will create a separate issue in `openforcefields` to ensure these additions are propagated there.

**Comment by dotsdl** [comment]
@j-wags beat me to it. :grin:
